app/Http/Controllers/Table/LocationController.php

```diff
@@ -44,7 +44,16 @@ public function searchFields($request)
 
     protected function sortFields($request)
     {
-        return ['location', 'devices', 'network', 'servers', 'firewalls', 'down'];
+        $device_types = [];
+        foreach (\LibreNMS\Config::get('device_types') as $device_type) {
```

Comment:
This can be simplified to:
```php
$device_types = array_column(\LibreNMS\Config::get('device_types'), 'type');

return array_merge(['location', 'devices', $device_types, ['down']);
```